src/wasm/wasm-type.cpp

```diff
@@ -470,42 +470,53 @@ std::optional<HeapType> getBasicHeapTypeLUB(HeapType::BasicHeapType a,
   if (unsigned(a) > unsigned(b)) {
```

Comment:
I would expect to see `if a.shared != b.shared return {}` since I assume the LUB of `func` and `(shared func)` is empty. Is that not so?

Instead IIUC this computes the LUB on the unshared variants of them both, then makes that LUB shared if `a` is. Why is that correct? (and why is it ok to pick the shareability from `a` specifically?)

Reply:
That's covered by the `HeapType(a).getTop() != HeapType(b).getTop()` check up above. Shared and unshared types will always have different top types.

Reply:
Ah, right, thanks!